packages/mobile/src/screens/search-screen-v2/RecentSearches.tsx

```diff
@@ -1,20 +1,241 @@
+import { useCallback } from 'react'
+
+import {
+  useGetPlaylistById,
+  useGetTrackById,
+  useGetUserById
+} from '@audius/common/api'
 import { recentSearchMessages as messages } from '@audius/common/messages'
-import { searchSelectors } from '@audius/common/store'
-import { useSelector } from 'react-redux'
+import { Kind, SquareSizes, Status } from '@audius/common/models'
+import { searchActions, searchSelectors } from '@audius/common/store'
+import type { SearchItem } from '@audius/common/store'
+import { profilePage } from '@audius/web/src/utils/route'
+import { Link } from '@react-navigation/native'
+import type { To } from '@react-navigation/native/lib/typescript/src/useLinkTo'
+import { useDispatch, useSelector } from 'react-redux'
+
+import {
+  Button,
+  Flex,
+  IconButton,
+  IconClose,
+  Text,
+  spacing,
+  useTheme
+} from '@audius/harmony-native'
+import { FlatList, ProfilePicture } from 'app/components/core'
+import { CollectionImageV2 } from 'app/components/image/CollectionImageV2'
+import { TrackImageV2 } from 'app/components/image/TrackImageV2'
+import Skeleton from 'app/components/skeleton'
+import { UserLink } from 'app/components/user-link'
 
-import { Flex, Text } from '@audius/harmony-native'
+import type { AppTabScreenParamList } from '../app-screen'
 
 const { getV2SearchHistory } = searchSelectors
+const { removeItem, clearHistory } = searchActions
+
+type RecentSearchProps = {
+  children: React.ReactNode
+  to: To<AppTabScreenParamList>
+  searchItem: SearchItem
+}
+
+const RecentSearch = (props: RecentSearchProps) => {
+  const { children, to, searchItem } = props
+  const dispatch = useDispatch()
+
+  const handleClickRemove = useCallback(() => {
+    dispatch(removeItem({ searchItem }))
+  }, [dispatch, searchItem])
+
+  return (
+    <Link to={to}>
+      <Flex
+        direction='row'
+        w='100%'
+        justifyContent='space-between'
+        alignItems='center'
+        pv='s'
+      >
+        <Flex direction='row' gap='m'>
+          {children}
+        </Flex>
+        <IconButton
+          aria-label={messages.remove}
+          icon={IconClose}
+          color='subdued'
+          size='s'
```

Comment:
Size `s` does seem correct according to the designs but in your screenshot the close buttons look slightly big. Can we double check they're the correct size?

Reply:
right? i think there is something up with button sizes because the big button refactor looks weird, will do.